tests/functional/src/main/java/org/mule/functional/classloading/isolation/classification/ClassPathClassifierContext.java

```diff
@@ -0,0 +1,234 @@
+import java.net.URL;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Represents a context that contains what is needed in order to do a classpath classification.
+ * It is used in {@link ClassPathClassifier}.
+ *
+ * @since 4.0
+ */
+public class ClassPathClassifierContext
+{
+    public static final int GROUP_ID_ARTIFACT_ID_TYPE_PATTERN_CHUNKS = 3;
+
+    private final Class<?> testClass;
+    private final List<URL> classPathURLs;
+    private final LinkedHashMap<MavenArtifact, Set<MavenArtifact>> mavenDependencies;
+    private final MavenMultiModuleArtifactMapping mavenMultiModuleArtifactMapping;
+    private final Predicate<MavenArtifact> exclusions;
+    private final Set<String> extraBootPackages;
+
+    protected final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    /**
+     * Creates a context used for doing the classification of the class path.
+     *
+     * @param testClass the test {@link Class} being tested
+     * @param classPathURLs the whole set of {@link URL}s that were loaded by IDE/Maven Surefire plugin when running the test
+     * @param mavenDependencies the tree of maven dependencies for the artifact that the test belongs to
+     * @param mavenMultiModuleArtifactMapping a mapper to get multi-module folder for artifactIds
+     */
+    public ClassPathClassifierContext(final Class<?> testClass, final List<URL> classPathURLs, final LinkedHashMap<MavenArtifact, Set<MavenArtifact>> mavenDependencies, final MavenMultiModuleArtifactMapping mavenMultiModuleArtifactMapping) throws IOException
+    {
+        this.testClass = testClass;
+        this.classPathURLs = classPathURLs;
+        this.mavenDependencies = mavenDependencies;
+        this.mavenMultiModuleArtifactMapping = mavenMultiModuleArtifactMapping;
+
+        Properties excludedProperties = getExcludedProperties();
+        this.exclusions = createExclusionsPredicate(testClass, excludedProperties);
+        this.extraBootPackages = getExtraBootPackages(testClass, excludedProperties);
+    }
+
+    /**
+     * @return the {@link Class} for the test that is going to be executed
+     */
+    public Class<?> getTestClass()
+    {
+        return testClass;
+    }
+
+    /**
+     * @return a {@link List<URL>} of URLs for the classpath provided by JUnit (it is the complete list of URLs)
+     */
+    public List<URL> getClassPathURLs()
+    {
+        return classPathURLs;
+    }
+
+    /**
+     * @return a {@link LinkedHashMap< MavenArtifact , Set<MavenArtifact>>} Maven dependencies for the given artifact
+     * tested (with its duplications). The map has as key an artifact and values are its dependencies. The first key in
+     * the map represents the root of the dependencies tree.
+     */
+    public LinkedHashMap<MavenArtifact, Set<MavenArtifact>> getMavenDependencies()
```

Comment:
The dependencies map should follow the contract of having as first key the
current artifact that's why I thought that it will be better to enforce
that by also having a LinkedHashMap so I know that it is a Map that has a
predictable iteration order. The other way would be to create another
entity that represents the node and its dependencies and can be traverse as
I do currently with the Map.

On Mon, Jul 11, 2016 at 6:31 PM, Pablo Kraan <notifications@github.com>
wrote:

> In
> tests/functional/src/main/java/org/mule/functional/classloading/isolation/classification/ClassPathClassifierContext.java
> <https://github.com/mulesoft/mule/pull/2930#discussion_r70341181>:
>
> > +    }
> > +
> > +    /**
> > +     * @return a {@link List<URL>} of URLs for the classpath provided by JUnit (it is the complete list of URLs)
> > +     */
> > +    public List<URL> getClassPathURLs()
> > +    {
> > +        return classPathURLs;
> > +    }
> > +
> > +    /**
> > +     * @return a {@link LinkedHashMap< MavenArtifact , Set<MavenArtifact>>} Maven dependencies for the given artifact
> > +     * tested (with its duplications). The map has as key an artifact and values are its dependencies. The first key in
> > +     * the map represents the root of the dependencies tree.
> > +     */
> > +    public LinkedHashMap<MavenArtifact, Set<MavenArtifact>> getMavenDependencies()
>
> Can this method return a Map instead?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mulesoft/mule/pull/2930/files/2679e5f7d162dc3ef827ce8f652c42c41a2656e9#r70341181>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AG96KWZ9a0KLiGaUulbEzQjoM3jnMHMiks5qUrZKgaJpZM4JGYaT>
> .
>



-- 
<http://www.mulesoft.com/>
Guillermo Fernandes, Senior Software Engineer

Av. Corrientes 316 Entre Piso, Buenos Aires, C1043AAQ, Argentina


<https://www.linkedin.com/company/www.mulesoft.com>
<https://twitter.com/mulesoft>   <https://www.facebook.com/MuleSoft>
<https://www.youtube.com/user/mulesoftvids> We're hiring!
<http://www.mulesoft.com/careers>